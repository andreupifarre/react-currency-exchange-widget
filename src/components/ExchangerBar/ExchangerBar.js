import React from 'react';
import PropTypes from 'prop-types';

import './ExchangerBar.scss';

const ExchangerBar = ({ onSwap, fromSign, toSign, rate }) => {
  const formatRate = rate && rate.toFixed(4);

  return (
    <div className="exchanger-bar">
      <button
        className="exchanger-bar__button exchanger-bar__button--swap"
        onClick={onSwap}
      />
      <button className="exchanger-bar__button exchanger-bar__button--rates">
        {fromSign}1 = {toSign}
        {formatRate}
      </button>
      <div className="exchanger-bar__empty-action" />
    </div>
  );
};

export default ExchangerBar;

ExchangerBar.defaultProps = {
  fromSign: '',
  toSign: '',
  rate: null,
  onSwap: () => {},
};

ExchangerBar.propTypes = {
  fromSign: PropTypes.string.isRequired,
  toSign: PropTypes.string.isRequired,
  rate: PropTypes.number,
  onSwap: PropTypes.func.isRequired,
};
