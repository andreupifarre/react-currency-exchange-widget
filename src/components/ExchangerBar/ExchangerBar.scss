.exchanger-bar {
  height: 42px;
  transform: translateY(-50%);
  margin-left: -20px;
  margin-right: -20px;
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: space-between;

  &__button {
    height: inherit;
    border-radius: 25px;
    border: 0;
    border: 2px solid #f3f4f5;
    min-width: 42px;
    background-color: white;
    color: #0075eb;
    font-size: 19px;
    padding: 0 17px;
    line-height: 19px;
    cursor: pointer;
    background-repeat: no-repeat;
    background-size: 18px;
    background-position: 10px center;

    &--swap {
      background-image: url(/img/swap-icon.png);
    }

    &--rates {
      background-image: url(/img/rates-icon.png);
      padding-left: 40px;
    }

    &:hover {
      border-color: #0075eb;
    }

    &:focus {
      outline: 0;
    }
  }

  &__empty-action {
    width: 42px;
  }
}
