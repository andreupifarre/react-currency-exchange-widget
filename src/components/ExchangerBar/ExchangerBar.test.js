import React from 'react';
import { shallow } from 'enzyme';

import ExchangerBar from './ExchangerBar';

const render = (props) => shallow(<ExchangerBar {...props} />);

describe('ExchangerBar', () => {
  it('should contain a parent element with a class .exchanger-bar', () => {
    const element = render().find('.exchanger-bar');
    expect(element.exists()).toBeTruthy();
  });

  it('should contain a button element with a class .exchanger-bar__button--swap', () => {
    const element = render().find('.exchanger-bar__button--swap');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('button');
  });

  it('should contain a button element with a class .exchanger-bar__button--rates', () => {
    const element = render().find('.exchanger-bar__button--rates');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('button');
  });

  it('should render correctly', () => {
    expect(render().getElements()).toMatchSnapshot();
  });
});
