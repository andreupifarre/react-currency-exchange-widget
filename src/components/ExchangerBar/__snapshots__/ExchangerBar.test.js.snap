// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangerBar should render correctly 1`] = `
Array [
  <div
    className="exchanger-bar"
  >
    <button
      className="exchanger-bar__button exchanger-bar__button--swap"
      onClick={[Function]}
    />
    <button
      className="exchanger-bar__button exchanger-bar__button--rates"
    >
      
      1 = 
      
    </button>
    <div
      className="exchanger-bar__empty-action"
    />
  </div>,
]
`;
